src/redux/features/admin/deactivateUserAcountSlice.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable prettier/prettier */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+import API from '@/api/api';
+import isAuth from '@/helpers/isAuth';
+
+export const deactivateUserAcount = createAsyncThunk(
+  'disable/user',
+  async (data: string) => {
+    const body={
+      email:data,
+    }
+    const { data: deactivateUser } = await API.patch('/users/deactivate',body, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+    return deactivateUser;
+  },
+);
+
+interface InitialState {
+  user: any | null;
+  isLoading: boolean;
+  error: string | null;
+}
+
+const isAuthData = isAuth();
```

Comment:
Consider the comments on the `activateUserAcount` function and make some improvements here.